src/main/java/com/limechain/rpc/methods/state/StateRPCImpl.java

```diff
@@ -255,29 +254,29 @@ public List<StorageChangeSet> stateQueryStorage(final List<String> keysHex, fina
     /**
      * Generates a read proof for a set of keys, scoped to a specific block.
      *
-     * @param keyHexList A list of keys in hexadecimal format for which the read proof is generated.
+     * @param keyHexList   A list of keys in hexadecimal format for which the read proof is generated.
      * @param blockHashHex The block hash in hexadecimal format to scope the query.
      * @return A map containing the block hash and the read proof for the specified keys within that block.
      */
     public Map<String, Object> stateGetReadProof(final List<String> keyHexList, final String blockHashHex) {
         final Hash256 blockHash = getHash256FromHex(blockHashHex);
 
         final byte[] blockStateRoot = blockState.getBlockStateRoot(blockHash).getBytes();
-        BlockTrieAccessor blockTrieAccessor = new BlockTrieAccessor(trieStorage, blockStateRoot);
 
         List<String> readProof = keyHexList
-                .stream()
-                .map(StringUtils::hexToBytes)
-                .map(key -> blockTrieAccessor
-                        .findMerkleValue(Nibbles.fromBytes(key))
-                        .orElse(null))
-                .filter(Objects::nonNull)
-                .map(StringUtils::toHexWithPrefix)
-                .toList();
+            .stream()
+            .map(StringUtils::hexToBytes)
+            // TODO change implementation logic as this is not correct
```

Comment:
is this comment outside of the scope of the PR?

Reply:
Indeed it is. I made changes so that it's not tightly coupled to the memory trie accessor.